app/src/ui/add-repository/add-existing-repository.tsx

```diff
@@ -18,28 +22,49 @@ interface IAddExistingRepositoryProps {
 
 interface IAddExistingRepositoryState {
   readonly path: string
-  readonly isGitRepository: boolean | null
+
+  /** Specifies if the path given is a git repository */
+  readonly isGitRepository: boolean
+
+  /** Flag used to display a warning that the given path is not a git repository */
+  readonly showNonGitRepositoryWarning: boolean
 }
 
-/** The component for adding or initializing a new local repository. */
+/** The component for adding a new local repository. */
 export class AddExistingRepository extends React.Component<IAddExistingRepositoryProps, IAddExistingRepositoryState> {
   private checkGitRepositoryToken = 0
 
   public constructor(props: IAddExistingRepositoryProps) {
     super(props)
 
-    this.state = { path: '', isGitRepository: false }
+    this.state = {
+      path: '',
+      isGitRepository: false,
+      showNonGitRepositoryWarning: false,
+    }
   }
 
-  public render() {
-    const disabled = this.state.path.length === 0 || this.state.isGitRepository == null
+  private renderWarning() {
+    if (!this.state.path.length || !this.state.showNonGitRepositoryWarning) {
+      return null
+    }
+
+    return (
+      <Row className='warning-helper-text'>
+        <Octicon symbol={OcticonSymbol.alert} />
+        <p>
+          This directory does not appear to be a git repository. Would you like to <LinkButton onClick={this.onCreateRepositoryClicked}>create a repository</LinkButton> here instead?
+        </p>
+      </Row>
+    )
+  }
 
-    const submitButtonText = this.state.isGitRepository
-      ? (__DARWIN__ ? 'Add Repository' : 'Add repository')
-      : (__DARWIN__ ? 'Create & Add Repository' : 'Create & add repository')
+  public render() {
+    const disabled = this.state.path.length === 0 || this.state.isGitRepository === null || !this.state.isGitRepository
```

Comment:
Since we've changed the `isGitRepository` state field to not be nullable I think we can drop the second condition here